lib/grammers-tl-types/tl/api.tl

```diff
@@ -1,32 +1,4 @@
-///////////////////////////////
-/////////////////// Layer cons
-///////////////////////////////
-
-//invokeAfterMsg#cb9f372d msg_id:long query:!X = X;
-//invokeAfterMsgs#3dc4b4f0 msg_ids:Vector<long> query:!X = X;
-//invokeWithLayer1#53835315 query:!X = X;
-//invokeWithLayer2#289dd1f6 query:!X = X;
-//invokeWithLayer3#b7475268 query:!X = X;
-//invokeWithLayer4#dea0d430 query:!X = X;
-//invokeWithLayer5#417a57ae query:!X = X;
-//invokeWithLayer6#3a64d54d query:!X = X;
-//invokeWithLayer7#a5be56d3 query:!X = X;
-//invokeWithLayer8#e9abd9fd query:!X = X;
-//invokeWithLayer9#76715a63 query:!X = X;
-//invokeWithLayer10#39620c41 query:!X = X;
-//invokeWithLayer11#a6b88fdf query:!X = X;
-//invokeWithLayer12#dda60d3c query:!X = X;
-//invokeWithLayer13#427c8ea2 query:!X = X;
-//invokeWithLayer14#2b9b08fa query:!X = X;
-//invokeWithLayer15#b4418b64 query:!X = X;
-//invokeWithLayer16#cf5f0987 query:!X = X;
-//invokeWithLayer17#50858a19 query:!X = X;
-//invokeWithLayer18#1c900537 query:!X = X;
-//invokeWithLayer#da9b0d0d layer:int query:!X = X; // after 18 layer
-
-///////////////////////////////
-///////// Main application API
-///////////////////////////////
+// LAYER 143
```

Comment:
Where's this TL from, tdesktop?

Reply:
> Where's this TL from, tdesktop?

https://github.com/tdlib/td 
tdesktop should be using that lib too

Reply:
What are drawbacks of putting `generated.rs` into src and committing to git? And manually re-generating it when schema is updated. This way RA will work fine and no double-compilation happens.

Reply:
> > Where's this TL from, tdesktop?
> 
> https://github.com/tdlib/td tdesktop should be using that lib too

I mean, I think I've always grabbed it from tdesktop (main reason being so the order of the definitions doesn't change much, making the diffs harder). The layer comment at the top made me ask this since tdesktop has it at the bottom.

> What are drawbacks of putting `generated.rs` into src and committing to git? And manually re-generating it when schema is updated. This way RA will work fine and no double-compilation happens.

I don't feel all that comfortable writing outside of the designated `OUT_DIR`. Committing to git would also be a large unnecessary bloat. In any case this topic needs a separate issue or PR to talk about.